apps/event-worker/src/target-scan/eks.ts

```diff
@@ -0,0 +1,176 @@
+import type { Cluster, EKSClient } from "@aws-sdk/client-eks";
+import type { STSClient } from "@aws-sdk/client-sts";
+import type { ResourceProviderAws, Workspace } from "@ctrlplane/db/schema";
+import type { KubernetesClusterAPIV1 } from "@ctrlplane/validators/resources";
+import { DescribeRegionsCommand } from "@aws-sdk/client-ec2";
+import {
+  DescribeClusterCommand,
+  ListClustersCommand,
+} from "@aws-sdk/client-eks";
+import _ from "lodash";
+import { isPresent } from "ts-is-present";
+
+import { logger } from "@ctrlplane/logger";
+import { ReservedMetadataKey } from "@ctrlplane/validators/conditions";
+
+import type { AwsCredentials } from "./aws.js";
+import { omitNullUndefined } from "../utils.js";
+import { assumeRole, assumeWorkspaceRole } from "./aws.js";
+
+const log = logger.child({ label: "resource-scan/eks" });
+
+const convertEksClusterToKubernetesResource = (
+  accountId: string,
+  cluster: Cluster,
+): KubernetesClusterAPIV1 => {
+  const region = cluster.endpoint?.split(".")[2];
+  const appUrl = `https://console.aws.amazon.com/eks/home?region=${region}#/clusters/${cluster.name}`;
```

Comment:
what about gov cloud? how does this work? can we not generate it from the cluster object?

Reply:
the ARN is really the only reliable source for this information. The ARN will contain the partition information, which is why we're using it as the primary source and falling back to region-based detection.

Commercial: `arn:aws:eks:region:account-id:cluster/cluster-name`
GovCloud: `arn:aws-us-gov:eks:region:account-id:cluster/cluster-name`


```tsx
 const partition = cluster.arn?.split(":")[1] ?? 
    (region?.startsWith("us-gov-") ? "aws-us-gov" : "aws");
  
  const appUrl = `https://${
    partition === "aws-us-gov" 
      ? `console.${region}.${partition}`
      : "console.aws.amazon"
  }.com/eks/home?region=${region}#/clusters/${cluster.name}`;
  ```